### ISSUE TITLE
The logstash-core-api need to be build on jruby

### ISSUE THREAD

**Issue by oyld** [elastic/logstash#5345]
I wrote a output plugin and run bundle install, but an error occurred:

```
There was a RuntimeError while loading logstash-core-plugin-api.gemspec:
The logstash-core-api need to be build on jruby from
  /home/ouyangliduo/.bundler/ruby/1.8/logstash-7cc3171676d7/logstash-core-plugin-api/logstash-core-plugin-api.gemspec:27
```

The variable RUBY_PLATFORM is x86_64-linux.

**Comment by suyograo** [comment]
What is your `ruby -v`. We currently only support running Logstash on JRuby

**Comment by oyld** [comment]
ruby 1.8.7 (2013-06-27 patchlevel 374) [x86_64-linux]
I have jruby and I can run logstash.

**Comment by suyograo** [comment]
@oyld can you switch to jruby? You can use something like [`rvm`](https://rvm.io/) to manage mutiple rubies on your machine. LS can only be built on jruby.

**Comment by suyograo** [comment]
We recommend using jruby 1.7.25

**Comment by oyld** [comment]
I did what you asked. But there is an other error:
```
Could not find gem 'logstash java' in git://github.com/elastic/logstash.git (at
2.3@7cc3171).
Source does not contain any versions of 'logstash java'
```

**Comment by oyld** [comment]
I solved it.
The [document](https://www.elastic.co/guide/en/logstash/current/_how_to_write_a_logstash_output_plugin.html#_add_a_literal_gemspec_literal_file_4) says Gemfile is
```
source 'https://rubygems.org'
gemspec
gem "logstash", :github => "elastic/logstash", :branch => "2.3"
```

I change it to
```
source 'https://rubygems.org'
gemspec
```

**Comment by spuder** [comment]
You sure removing that line is good advice? [From the documentation](https://www.elastic.co/guide/en/logstash/current/_how_to_write_a_logstash_input_plugin.html#_add_a_gemfile) it sounds like that line might be important. 

>Gemfiles allow Ruby’s Bundler to maintain the dependencies for your plugin. Currently, all we’ll need is the Logstash gem, for testing, but if you require other gems, you should add them in here.

**Comment by z-vr** [comment]
I can't test my plugin, 
```
computer:logstash-input-myplugin zavr$ ruby -v
jruby 1.7.23 (1.9.3p551) 2015-11-24 f496dd5 on Java HotSpot(TM) 64-Bit Server VM 1.8.0_112-b16 +jit [darwin-x86_64]
```
```
computer:logstash-input-myplugin zavr$ which bundle
/Users/zavr/.rvm/gems/jruby-1.7.23/bin/bundle
```
```
Could not find gem 'logstash java' in git://github.com/elastic/logstash.git (at
5.1@5a16cb3).
Source does not contain any versions of 'logstash java'
```

**Comment by andrewvc** [comment]
@z-vr that's a separate issue. Try running a bundle update, it should be resolved now with logstash-core-5.2.1-snapshot1

**Comment by caperea** [comment]
@z-vr was your issue with "Source does not contain any versions of 'logstash java'" resolved? How?
